File: CS50,2025-26/148145442-main/148145442-main/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int cents;

    // Prompt user until a non-negative integer is entered
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    int coins = 0;

    // Use greedy algorithm to calculate minimum coins
    coins += cents / 25;
    cents %= 25;

    coins += cents / 10;
    cents %= 10;

    coins += cents / 5;
    cents %= 5;

    coins += cents; // Remaining pennies

    // Output result
    printf("%d\n", coins);
}

File: CS50,2025-26/148145442-main/148145442-main/sentimental-cash/cash.py
# sentimental-cash/cash.py

from cs50 import get_float

def main():
    # Prompt user until a non-negative value is entered
    while True:
        try:
            dollars = get_float("Change owed: ")
            if dollars >= 0:
                break
        except:
            continue

    # Convert dollars to cents and round to avoid floating-point imprecision
    cents = round(dollars * 100)

    # Initialize coin counter
    coins = 0

    # Calculate minimum number of coins
    for coin in [25, 10, 5, 1]:
        coins += cents // coin
        cents %= coin

    # Output the result
    print(coins)

if __name__ == "__main__":
    main()
